src/Core/src/Platform/Android/ButtonExtensions.cs

```diff
@@ -6,22 +6,50 @@ namespace Microsoft.Maui.Platform
 {
 	public static class ButtonExtensions
 	{
-		public static void UpdateStrokeColor(this MaterialButton platformButton, IButtonStroke buttonStroke)
+		public static void UpdateBackground(this MaterialButton platformView, IButton button)
 		{
-			if (buttonStroke.StrokeColor is Color stroke)
-				platformButton.StrokeColor = ColorStateListExtensions.CreateButton(stroke.ToPlatform());
+			var background = button.Background;
+
+			if (background is SolidPaint)
+				platformView.UpdateBackground(background);
+			else
+				platformView.UpdateBorderDrawable(button);
```

Comment:
Why do we have this issue? Basically because we have implemented Handlers, in this specific case the Button in different phases and we have not detected this problem with tests. We initially implement the Background property using an **AppCompatButton**. Subsequently, by implementing properties related to the Border (BorderWidth, etc) we change the type of native control to use for a **MaterialButton**.

The MaterialButton requires **BackgroundTintList** to be set to null in order to render a custom background. If a custom background is set, a Drawable is used where the background is drawn and, if there is a border, use the Drawabable too. Otherwise, the BackgroundColor property and the StrokeWidth etc properties of the MaterialButton itself are used.